### ISSUE TITLE
Feature Request: Responsive uk-grid gutter (i.e. uk-grid-medium / large)

### ISSUE THREAD

**Issue by paul-muckypuddle** [uikit/uikit#3411]
It'd be really handy to have the uk-grid-collapse / small / medium / large classes work responsively like grid width. For example

~~~~
<div uk-grid class="uk-child-width-1-3@l uk-child-width-1-1 uk-grid-medium@l uk-grid-small">
    <div>Foo</div>
    <div>Bar</div>
    <div>Baz</div>
</div>
~~~~

This would help making 'tighter' sites on small devices, without the same constriction on larger desktops, much easier.

**Comment by zzseba78** [comment]
The default uikit grid it´s dinamic, so in default, small and medium have a smaller gutter, and from large the gutter it´s larger. Of  course you can customize this gutter values in less / sass.
https://github.com/uikit/uikit/blob/067028fe54f4109b33a890a50e45d53449eca860/src/less/components/grid.less#L29-L32

**Comment by paul-muckypuddle** [comment]
Ah I didn't realise this, thanks!

I think then it's making this more granular - the main tasks they'd be handy for is mobile landscape / portrait and tablet landscape / portrait, in which the resolution jumps are relatively small but space is at a premium, and gutters on a portrait phone should ideally be smaller than a landscape tablet.

**Comment by zzseba78** [comment]
Yes, it´s a good idea, but i guess the idea is to keep uikit lightweight and not bloat code, and this is for a very specific use case, for example, in your task this can be done with default grid ( remember that large grid it´s dinamic too, so you have another sizing alternative ).

**Comment by paul-muckypuddle** [comment]
I do agree on bloat and I'm glad Uikit is as tightly constrained as it is for the most part, but not sure I agree on thins being a 'very specific use case' - having equal gutters on 375 x 667 and 1024 x 768 but being able to vary them between 1024 x 768 vs e.g. 1920 x 1080 seems like having priorities the wrong way round.

**Comment by zzseba78** [comment]
Remember you can customize breakpoing points to your needs.
